src/codegen/compile.ml

```diff
@@ -24,6 +24,8 @@ let (@@) = Stdlib.(@@)
 module G = InstrList
 let (^^) = G.(^^) (* is this how we import a single operator from a module that we otherwise use qualified? *)
 
+let enable_write_barrier = true
```

Comment:
I wonder if it would make sense to just make this enabled only when sanity checking for the compacting or copying_gc and always for the generational gc. Then you could actually check it in (not just have it as an experiment).

I support there would still be the overhead of testing whether the check needs to done during collection, so perhaps unacceptable for the non-generational gcs...